build/images/codegen/Dockerfile

```diff
@@ -48,7 +48,7 @@ RUN go install k8s.io/code-generator/cmd/client-gen@kubernetes-$K8S_VERSION && \
     go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.33.0 && \
     go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.3.0 && \
     go install golang.org/x/tools/cmd/goimports@latest && \
-    go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
+    go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.3
```

Comment:
@antoninbas Do we need to update the codegen image because of this change?

Reply:
My bad. I must have lost that change when I reorganized commits in the PR before submitting it for review.

Yes, we should update the codegen image. I have pushed `antrea/codegen:kubernetes-1.31.1-build.1`:
```
C02F16ASMD6R:codegen abas$ docker run -ti antrea/codegen:kubernetes-1.31.1-build.1 bash
root@cbc09d4c9e99:/go# controller-gen --version
Version: v0.16.3
```

Please add a line in the README:
```
| kubernetes-1.31.1-build.1 | Upgraded controller-gen to v0.16.3
```
And edit the one for `kubernetes-1.31.1` to remove the invalid controller-gen mention

I checked and it seems updating to controller-gen v0.16.3 didn't have an impact to generated fields. I am not sure which files are generated this way, is it just `multicluster/apis/multicluster/v1alpha1/zz_generated.deepcopy.go` and `multicluster/apis/multicluster/v1alpha2/zz_generated.deepcopy.go`?

Reply:
Updated.
Yes, multicluster codes generation will use this tool.